### PR TITLE
APP-1653 : Show message when room was not created

### PR DESCRIPTION
**Pull request by crepache**

### CODE REVIEW

File: helpdesk-bot/src/main/java/org/symphonyoss/symphony/bots/helpdesk/bot/ticket/AcceptTicketService.java [hunk 1]
```diff
@@ -65,6 +65,7 @@ protected TicketResponse execute(Ticket ticket, SymUser agent) {
       try {
         updateMembership(agent.getId());
         addAgentToServiceStream(ticket, agent.getId());
+        sendMessageWithShowHistoryFalse(ticket, agent.getId());
```
Comment (robson-sanchez) [inline helpdesk-bot/src/main/java/org/symphonyoss/symphony/bots/helpdesk/bot/ticket/AcceptTicketService.java]: You should rename this method to 'sendTicketHistory'. You might also declare a variable to hold the agent id.

File: helpdesk-bot/src/main/java/org/symphonyoss/symphony/bots/helpdesk/bot/ticket/TicketService.java [hunk 2]
```diff
@@ -106,4 +131,42 @@ protected TicketResponse buildResponse(Ticket ticket, SymUser agent, String mess
 
   protected abstract TicketResponse execute(Ticket ticket, SymUser agentUser);
 
+  protected void sendMessageWithShowHistoryFalse(Ticket ticket, Long agentId) {
```
Comment (robson-sanchez) [inline helpdesk-bot/src/main/java/org/symphonyoss/symphony/bots/helpdesk/bot/ticket/TicketService.java]: This method might be moved to AcceptTicketService subclass once only this component requires this behavior. Then, you should try to reuse the method to send message to client stream.

File: helpdesk-bot/src/main/java/org/symphonyoss/symphony/bots/helpdesk/bot/ticket/TicketService.java [hunk 3]
```diff
@@ -106,4 +131,42 @@ protected TicketResponse buildResponse(Ticket ticket, SymUser agent, String mess
...
+        List<SymMessage> messages = symphonyClientUtil.getSymMessages(clientStream,
+            ticket.getQuestionTimestamp(), 100);
+
+        messages.stream()
+            .filter(symMessage -> ticket.getClient()
+                .getUserId()
+                .equals(symMessage.getSymUser().getId()))
+            .sorted(Comparator.comparing(SymMessage::getTimestamp))
+            .forEach(symMessage -> {
+              symMessage.setStream(serviceStream);
```
Comment (robson-sanchez) [inline helpdesk-bot/src/main/java/org/symphonyoss/symphony/bots/helpdesk/bot/ticket/TicketService.java]: Please use the 'map' method to modify the message object.

```
.sorted(Comparator.comparing(SymMessage::getTimestamp))
.map(symMessage -> {
  symMessage.setStream(serviceStream);
  symMessage.setStreamId(serviceStream.getStreamId());
  return symMessage;
})
.forEach(symMessage -> {
  sendMessage(symMessage, agentId);
});
```

File: helpdesk-bot/src/main/java/org/symphonyoss/symphony/bots/helpdesk/bot/ticket/TicketService.java [hunk 4]
```diff
@@ -28,16 +47,22 @@
...
+  protected final SymphonyClientUtil symphonyClientUtil;
+
+  protected final HelpDeskAi helpDeskAi;
+
   private final ValidateMembershipService validateMembershipService;
 
   public TicketService(SymphonyValidationUtil symphonyValidationUtil, SymphonyClient symphonyClient,
       HelpDeskBotConfig helpDeskBotConfig, TicketClient ticketClient,
-      ValidateMembershipService validateMembershipService) {
+      ValidateMembershipService validateMembershipService, HelpDeskAi helpDeskAi) {
```
Comment (robson-sanchez) [inline helpdesk-bot/src/main/java/org/symphonyoss/symphony/bots/helpdesk/bot/ticket/TicketService.java]: I believe this new parameter no longer required.

File: helpdesk-message-proxy-service/src/main/java/org/symphonyoss/symphony/bots/helpdesk/messageproxy/service/TicketService.java [hunk 5]
```diff
@@ -157,4 +164,27 @@ public void sendIdleMessageToAgentStreamId(SymMessage message) {
...
+    try {
+      symphonyClient.getMessagesClient().sendMessage(symMessage.getStream(), symMessage);
+    } catch (MessagesException e) {
+      LOGGER.error("Could not send message to client stream ID: ", e);
+    }
+  }
+
+  private String parseTemplate(String templateFilename) {
+    StringBuilder message = new StringBuilder();
+    InputStream resource = getClass().getClassLoader().getResourceAsStream(templateFilename);
```
Comment (robson-sanchez) [inline helpdesk-message-proxy-service/src/main/java/org/symphonyoss/symphony/bots/helpdesk/messageproxy/service/TicketService.java]: I don't think you require an XML file to define this message. You can create a constant field in the class.

File: helpdesk-message-proxy-service/src/main/java/org/symphonyoss/symphony/bots/helpdesk/messageproxy/service/TicketService.java [hunk 6]
```diff
@@ -157,4 +164,27 @@ public void sendIdleMessageToAgentStreamId(SymMessage message) {
       LOGGER.error("Could not send ticket message to agent stream ID: ", e);
     }
   }
+
+  public void sendMessageWhenRoomCreationFails(SymMessage symMessage) {
```
Comment (robson-sanchez) [inline helpdesk-message-proxy-service/src/main/java/org/symphonyoss/symphony/bots/helpdesk/messageproxy/service/TicketService.java]: Please include javadoc on new methods and classes.

File: helpdesk-message-proxy-service/src/main/resources/roomNotCreated.xml [hunk 7]
```diff
@@ -0,0 +1,3 @@
+<messageML>
```
Comment (robson-sanchez) [inline helpdesk-message-proxy-service/src/main/resources/roomNotCreated.xml]: Probably, this file no longer required.

**Review by robson-sanchez — CHANGES_REQUESTED** [review]

File: helpdesk-message-proxy-service/src/main/resources/roomNotCreated.xml (hunk 7 above)
Comment (crepache) [reply helpdesk-message-proxy-service/src/main/resources/roomNotCreated.xml]: Done.

File: helpdesk-message-proxy-service/src/main/java/org/symphonyoss/symphony/bots/helpdesk/messageproxy/service/TicketService.java (hunk 5 above)
Comment (crepache) [reply helpdesk-message-proxy-service/src/main/java/org/symphonyoss/symphony/bots/helpdesk/messageproxy/service/TicketService.java]: Done.

File: helpdesk-bot/src/main/java/org/symphonyoss/symphony/bots/helpdesk/bot/ticket/AcceptTicketService.java (hunk 1 above)
Comment (crepache) [reply helpdesk-bot/src/main/java/org/symphonyoss/symphony/bots/helpdesk/bot/ticket/AcceptTicketService.java]: Done.

File: helpdesk-bot/src/main/java/org/symphonyoss/symphony/bots/helpdesk/bot/ticket/TicketService.java (hunk 2 above)
Comment (crepache) [reply helpdesk-bot/src/main/java/org/symphonyoss/symphony/bots/helpdesk/bot/ticket/TicketService.java]: Done.

File: helpdesk-bot/src/main/java/org/symphonyoss/symphony/bots/helpdesk/bot/ticket/TicketService.java (hunk 3 above)
Comment (crepache) [reply helpdesk-bot/src/main/java/org/symphonyoss/symphony/bots/helpdesk/bot/ticket/TicketService.java]: Done.

File: helpdesk-message-proxy-service/src/main/resources/roomNotCreated.xml (hunk 7 above)
Comment (crepache) [reply helpdesk-message-proxy-service/src/main/resources/roomNotCreated.xml]: Done.

File: helpdesk-bot/src/main/java/org/symphonyoss/symphony/bots/helpdesk/bot/ticket/TicketService.java (hunk 4 above)
Comment (crepache) [reply helpdesk-bot/src/main/java/org/symphonyoss/symphony/bots/helpdesk/bot/ticket/TicketService.java]: Done.

File: helpdesk-message-proxy-service/src/main/java/org/symphonyoss/symphony/bots/helpdesk/messageproxy/service/TicketService.java (hunk 6 above)
Comment (crepache) [reply helpdesk-message-proxy-service/src/main/java/org/symphonyoss/symphony/bots/helpdesk/messageproxy/service/TicketService.java]: Done.

**Review by robson-sanchez — APPROVED** [review]

**Pull request merged**